Sources/FoundationEssentials/String/String+Internals.swift

```diff
@@ -23,7 +23,12 @@ import Darwin
 import WinSDK
 
 extension String {
-    package func withNTPathRepresentation<Result>(_ body: (UnsafePointer<WCHAR>) throws -> Result) throws -> Result {
+    /// Invokes `body` with a resolved and potentially `\\?\`-prefixed version of the pointee,
+    /// to ensure long paths greater than 260 characters are handled correctly.
+    ///
+    /// - parameter preserveRelativePaths: Returns the original path without transforming through GetFullPathNameW + PathCchCanonicalizeEx, if the path is shorter than MAX_PATH.
+    /// - seealso: https://learn.microsoft.com/en-us/windows/win32/fileio/maximum-file-path-limitation
+    package func withNTPathRepresentation<Result>(preserveRelativePaths: Bool = false, _ body: (UnsafePointer<WCHAR>) throws -> Result) throws -> Result {
```

Comment:
I don't think that `preserveRelativePaths` is the right term here - this isn't just preserving a relative path, its allowing for un-extended paths. `permitWin32Paths` perhaps? (The 261 limitation is Win32, not NT, `\\?\` bypaasses the Win32 layer and passes the path right to the NT subsystem).

Reply:
Good point... I actually do want to preserve relative paths, because the CreateSymbolicLinkW will behave differently if lpTargetFileName is absolute vs relative. There's simply no way to specify an extended-length absolute path, since the extended length prefix requires fully qualifying a path. I changed the implementation to match the expectations.

Reply:
I think that `relative:` is a better label then.

Reply:
Done